### PR TITLE
fix: prettyPrint options - misleading error message with wrong option value

### LINKED ISSUE #1007: Misleading error message on wrong options value
Hey 👋 

When passing wrong option value (`ignore` as an array):

```js
pino({
  prettyPrint: {
    ignore: ['hostname']
  }
})
```

The error in console is:

```console
Error: Missing `pino-pretty` module: `pino-pretty` must be installed separately
```

I think it's quite misleading, would be nice to have an options error or smth.

pino - 6.11.2
pino-pretty - 4.7.1

Thanks

### PR DESCRIPTION
**Pull request by lamanabie**
Fixes #1007

### CODE REVIEW

File: test/pretty.test.js [hunk 1]
```diff
@@ -66,6 +66,13 @@ test('throws when prettyPrint is true but pino-pretty module is not installed',
   require.cache[require.resolve('pino-pretty')].exports = prettyFactory
 })
 
+test('throws when prettyPrint has invalid options', async ({ throws, equal }) => {
+  throws(() => pino({ prettyPrint: { ignore: ['hostname'] } }))
+  try { pino({ prettyPrint: { ignore: ['hostname'] } }) } catch ({ message }) {
+    equal(message, 'opts.ignore.split is not a function')
+  }
```
Comment (simoneb) [inline test/pretty.test.js:73]: this test could be rewritten in a slightly better way:

- if no exception is thrown, this test will pass. it should fail instead
- the second assertion implies the first, which is then redundant

**Review by simoneb — COMMENTED** [review]
Do we have a test to check that the missing pino-pretty error is thrown sometimes?

**Review by mcollina — CHANGES_REQUESTED** [review]
Code is good but the test can be improved as @simoneb said.

File: test/pretty.test.js (hunk 1 above)
Comment (lamanabie) [reply test/pretty.test.js:73]: I did try run test with correct prettyPrint options and test correctly failed when no exception is thrown 
```
 ✖ expected to throw

  test/pretty.test.js                                                                      
  68 |                                                                                     
  69 | test('throws when prettyPrint has invalid options', async ({ throws, equal }) => {  
> 70 |   throws(() => pino({ prettyPrint: { ignore: 'hostname' } }))                       
     | --^                                                                                 
  71 |   try { pino({ prettyPrint: { ignore: 'hostname' } }) } catch ({ message }) {       
  72 |     equal(message, 'opts.ignore.split is not a function')                           
  73 |   }                                                                                 

  test: test/pretty.test.js throws when prettyPrint has invalid options
```
Thats why is there that first assertion, that validate if exception is thrown
```javascript
 throws(() => pino({ prettyPrint: { ignore: ['hostname'] } }))
```
Second assertion validate correct error message

I write this test case similar to another test case in test/pretty.test.js, that handle this almost same way
```javascript
test('throws when prettyPrint is true but pino-pretty module is not installed', async ({ throws, equal }) => {
  // pino pretty *is* installed, and probably also cached, so rather than
  // messing with the filesystem the simplest way to generate a not found
  // error is to simulate it:
  const prettyFactory = require('pino-pretty')
  require.cache[require.resolve('pino-pretty')].exports = () => {
    throw Error('Cannot find module \'pino-pretty\'')
  }
  throws(() => pino({ prettyPrint: true }))
  try { pino({ prettyPrint: true }) } catch ({ message }) {
    equal(message, 'Missing `pino-pretty` module: `pino-pretty` must be installed separately')
  }

  require.cache[require.resolve('pino-pretty')].exports = prettyFactory
})
```

Comment (simoneb) [reply test/pretty.test.js:73]: you are right that the test fails if no exception is thrown the first assertion will fail, but if the error message is wrong the second will not fail. please fix that one.

Comment (lamanabie) [reply test/pretty.test.js:73]: I did run test with different expected message: 
```javascript
test('throws when prettyPrint has invalid options', async ({ throws, equal }) => {
  throws(() => pino({ prettyPrint: { ignore: ['hostname'] } }))
  try { pino({ prettyPrint: { ignore: ['hostname'] } }) } catch ({ message }) {
    equal(message, 'this will fail')
  }
})
```
Test failed because error message does not equal
```
 FAIL  test/pretty.test.js
 ✖ should be equal

  test/pretty.test.js                                                                   
  70 |   throws(() => pino({ prettyPrint: { ignore: ['hostname'] } }))                  
  71 |   try { pino({ prettyPrint: { ignore: ['hostname'] } }) } catch ({ message }) {  
> 72 |     equal(message, 'this will fail')                                             
     | ----^                                                                            
  73 |   }                                                                              
  74 | })                                                                               
  75 |        
```

I believe test case is written correctly and handle all edge cases.

Comment (simoneb) [inline test/pretty.test.js:73]: Yes, it's correct but poorly written. Try a more idiomatic way to write the same:

```suggestion
  throws(() => pino({ prettyPrint: { ignore: ['hostname'] } }), 'opts.ignore.split is not a function')
```
see https://node-tap.org/docs/api/asserts/#tthrowsfn-expectederror-message-extra

Comment (lamanabie) [reply test/pretty.test.js]: I did rewrite all throws in tests that did follow bad pattern.

**Review by simoneb — APPROVED** [review]
Thanks for improving the existing tests too!

**Comment by simoneb** [comment]
@mcollina please take a look. some tests have been changed to avoid a weird pattern of asserting an error would be thrown and then try to throw it manually just to assert on the error message

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]
LGTM
